syntax = "proto2";

package tenon.bls.protobuf;

message VerifyVecItem {
    optional bytes x_c0 = 1;
    optional bytes x_c1 = 2;
    optional bytes y_c0 = 3;
    optional bytes y_c1 = 4;
    optional bytes z_c0 = 5;
    optional bytes z_c1 = 6;
}

message VerifyVecBrdReq {
    repeated VerifyVecItem verify_vec = 1;
    optional bytes public_ip = 2;
    optional uint32 public_port = 3;
}

message VerifyVecBrdRes {
    optional bytes public_ip = 1;
    optional uint32 public_port = 2;
}

message SwapSecKeyReq {
    optional bytes sec_key = 1;
    optional uint32 sec_key_len = 2;
}

message AgainstParticipant {
    optional uint32 against_index = 1;
}

message FinishBroadcast {
	repeated uint64 bitmap = 1;
}

message BlsMessage {
    optional VerifyVecBrdReq verify_brd = 1;
    optional SwapSecKeyReq swap_req = 2;
    optional AgainstParticipant against_req = 3;
    optional FinishBroadcast finish_req = 4;
    optional VerifyVecBrdRes verify_res = 5;
    optional uint32 index = 6;
    optional bytes sign_ch = 7;
	optional bytes sign_res = 8;
    optional uint64 elect_height = 9;
}
